### PR TITLE
move serviceaccountController start function init into StartControllers

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
make code more readable and graceful
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by gmarek** [comment]
cc @kubernetes/sig-api-machinery-misc 
/unassign

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -432,9 +431,10 @@ func CreateControllerContext(s *options.CMServer, rootClientBuilder, clientBuild
 	return ctx, nil
 }
 
-func StartControllers(ctx ControllerContext, startSATokenController InitFunc, controllers map[string]InitFunc) error {
+func StartControllers(ctx ControllerContext, rootClientBuilder controller.ControllerClientBuilder, controllers map[string]InitFunc) error {
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: I really don't want leak a `rootClientBuilder` into `StartControllers`.  Also, forcing the init function shape, makes sure the the init code for the SAToken controller will look consistent to people.

**Comment by ncdc** [comment]
I think it looks better the way it is in master. `StartControllers` doesn't need `rootClientBuilder` - it just needs an init func to start the service account token controller, which is what we have in master.

**Comment by hzxuzhonghu** [comment]
But I have different opinion serviceAccountController is a kind of controllers just as others, they should all init in StartControllers

**Comment by liggitt** [comment]
I don't think moving the special case controller into the function is a net improvement

**Comment by ncdc** [comment]
> serviceAccountController is a kind of controllers just as others, they should all init in StartControllers

This is what the code in master currently does.

**Comment by hzxuzhonghu** [comment]
@ncdc Ok， I agree with you all. I will close this pr.

**Pull request closed without merging**